ehrapy/preprocessing/_outliers.py

```diff
@@ -0,0 +1,123 @@
+from __future__ import annotations
+
+import numpy as np
+import pandas as pd
+import scipy.stats.mstats
+from anndata import AnnData
+
+
+def winsorize(
+    adata: AnnData,
+    vars: str | list[str] = None,
+    obs_cols: str | list[str] = None,
+    limits: list[float] = None,
+    copy: bool = False,
+    **kwargs,
+) -> AnnData:
+    """Returns a Winsorized version of the input array.
+
+    The implementation is based on https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.mstats.winsorize.html
+
+    Args:
+        adata: AnnData object to winsorize
+        vars: The features to winsorize
+        obs_cols: Columns in obs with features to winsorize
+        limits: Tuple of the percentages to cut on each side of the array as floats between 0. and 1. (default: 0.01 and 0.99)
+        copy: Whether to return a copy or not
+        **kwargs: Keywords arguments get passed to scipy.stats.mstats.winsorize
+
+    Returns:
+        Winsorized AnnData object if copy is True
+    """
+    if vars is not None and not all(elem in adata.var_names.values for elem in vars):
+        raise ValueError(
+            f"Columns `{[col for col in vars if col not in adata.var_names.values]}` are not in var_names."
+        )
+
+    if obs_cols is not None and not all(elem in adata.obs.columns.values for elem in obs_cols):
+        raise ValueError(
+            f"Columns `{[col for col in obs_cols if col not in adata.obs.columns.values]}` are not in obs."
+        )
+
+    if isinstance(vars, str):  # pragma: no cover
+        vars = [vars]
+
+    if isinstance(obs_cols, str):  # pragma: no cover
+        obs_cols = [obs_cols]
+
+    if copy:  # pragma: no cover
+        adata = adata.copy()
+
+    if limits is None:
+        limits = [0.01, 0.99]
```

Comment:
move to default parameters

Reply:
Mutable datastructures should not be default parameters. This circumvents a flake8 issue (and is the correct way to fix this)